### PR TITLE
create property connectedTo if it doesn't exist

### PR DESCRIPTION
**Pull request by SimonLab**
see [issue 139](https://github.com/FAC-GM/app/issues/139)
see [issue 131](https://github.com/FAC-GM/app/issues/131)

This PR is ready for a review, @nelsonic

### CODE REVIEW

File: test/api.test.js [hunk 1]
```diff
@@ -148,6 +148,50 @@ describe('api /profile', function () {
     });
   });
 
+  it("create a profile which doesn't have the property connectedTo", function (done) {
```
Comment (nelsonic) [inline test/api.test.js]: nice to see regression tests being added. :+1:

File: views/partials/candidate.html [hunk 2]
```diff
@@ -17,7 +17,9 @@
 
     {{#each connectedTo}}
       <ul>
-        <li><span class='initials'>{{initials_of_connected_users this}}</span></li>
+        {{#if this}}
```
Comment (nelsonic) [inline views/partials/candidate.html]: curious, why would `this` be undefined?

File: lib/api.js [hunk 3]
```diff
@@ -23,7 +23,7 @@ exports.register = function (server, options, next) {
             if(response.hits.hits.length > 0) {
                 var id = response.hits.hits[0]._id;
 
-                var connectedUsers = response.hits.hits[0]._source.connectedTo;
+                var connectedUsers = response.hits.hits[0]._source.connectedTo || [];
```
Comment (SimonLab) [inline lib/api.js]: Without this condition the connnectedUsers can be undefined

File: views/partials/candidate.html (hunk 2 above)
Comment (SimonLab) [inline views/partials/candidate.html]: connectedTo can be undefined due to the line 26 of api.js (see my comment on this line). Unfortunately some of our data still have undefined and we need to keep this if condition at the moment

Comment (nelsonic) [inline views/partials/candidate.html]: Right, but doesn't handlebars simply ignore this statement completely if the `connectedTo` is not set? in which case, would you not want to do the `if` statement *before* the `each` to confirm that `connectedTo` is set?

File: lib/api.js (hunk 3 above)
Comment (nelsonic) [inline lib/api.js]: Understood. :+1:

File: views/partials/candidate.html (hunk 2 above)
Comment (SimonLab) [inline views/partials/candidate.html]: connectedTo is defined, and looks like this in some specific cases:
```
connectedTo = [undefined, undefined,...];
```
That's why we need the if to check the values inside the array of connectedTo

Comment (nelsonic) [inline views/partials/candidate.html]: Oh... that's bad data though... we need to mitigate this in the data insertion rather than in the template...

**Comment by nelsonic** [comment]
Merging.

**Comment by iteles** [comment]
@nelsonic @SimonLab Do we need to open a technical task for [this note](https://github.com/FAC-GM/app/pull/140#discussion_r46444146)?

**Pull request merged**